### ISSUE TITLE
Add restoreAll

### ISSUE THREAD

**Issue by satazor** [saintedlama/betray#2]
It would be really cool if betray had a `restoreAll` function so that we can add to our `afterEach` lifecyle function.

**Comment by saintedlama** [comment]
Good point. Will take a look on how to get that implemented...

**Comment by satazor** [comment]
At first glance you only need to:

- push the restore function into an array when a betray is added
- remove the function from an array when it is restored
- `restoreAll` would just iterate this array and call each function.

**Comment by saintedlama** [comment]
Quite a global approach - cause betrayed functions are kept in a global list...

Thinking about a track root level function that implements a tracking functionality. Something like `betray.track()` returning a tracking instance of betray implementing a restoreAll function. Thoughts?

**Comment by saintedlama** [comment]
Got something implemented - an additional `record` function is exposed to record betrayed functions

```
var betrayed = betray.record();
// Same API as betray
betrayed(math, 'throwError');

// et voila - restore all
betrayed.restoreAll();
```

**Comment by satazor** [comment]
Sounds ok to me, though not so easily as the global approach. I see that you published as 1.2.0, I will give it a try.

**Comment by satazor** [comment]
I've tried and it's working great:

```js
const betray = require('betray');

describe('something', () => {
    let betrayed;

    beforeEach(() => { betrayed = betray.record(); });
    afterEach(() => betrayed.restoreAll());

    it('should do something', () => {
        // Use betrayed instead of betray
    });
});
```

I guess the only thing missing is to document this new feature in the README.
